lib/charliecloud.py

```diff
@@ -1187,6 +1248,9 @@ def root_env():
    def manifest_for_download(self, image_ref):
       return self.download_cache // ("%s.manifest.json" % image_ref.for_path)
```

Comment:
I don't think so. The only time we get a different manifest is when we use the fat manifest to pull the target image's manifest by digest. We compare the hash of this manifest with that in storage, and if they don't match we overwrite the storage contents for that particular image.

Reply:
Aha. In that case we should save manifests with their hash in the filename, like blobs. This will let us know if the file in cache is what we want without reading its content.